### PR TITLE
[SPARK-21181] Release byteBuffers to suppress netty error messages

### PR DESCRIPTION
**Pull request by dhruve**
## What changes were proposed in this pull request?
We are explicitly calling release on the byteBuf's used to encode the string to Base64 to suppress the memory leak error message reported by netty. This is to make it less confusing for the user.

### Changes proposed in this fix
By explicitly invoking release on the byteBuf's we are decrement the internal reference counts for the wrappedByteBuf's. Now, when the GC kicks in, these would be reclaimed as before, just that netty wouldn't report any memory leak error messages as the internal ref. counts are now 0.

## How was this patch tested?
Ran a few spark-applications and examined the logs. The error message no longer appears.

Original PR was opened against branch-2.1 => https://github.com/apache/spark/pull/18392

### CODE REVIEW

File: common/network-common/src/main/java/org/apache/spark/network/sasl/SparkSaslServer.java [hunk 1]
```diff
@@ -187,14 +188,30 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
...
   public static char[] encodePassword(String password) {
     Preconditions.checkNotNull(password, "Password cannot be null if SASL is enabled");
-    return Base64.encode(Unpooled.wrappedBuffer(password.getBytes(StandardCharsets.UTF_8)))
-      .toString(StandardCharsets.UTF_8).toCharArray();
+    return getBase64EncodedString(password).toCharArray();
+  }
+
+  /** Return a Base64-encoded string. */
+  private static String getBase64EncodedString (String str) {
+    ByteBuf byteBuf = null, encodedByteBuf = null;
```
Comment (srowen) [inline common/network-common/src/main/java/org/apache/spark/network/sasl/SparkSaslServer.java]: Normally these would be on two lines -- think the linter will complain about this (but not fail the build, so take a look).
Also does it matter that they're not released in the reverse order they were obtained?

Comment (dhruve) [inline common/network-common/src/main/java/org/apache/spark/network/sasl/SparkSaslServer.java]: Okay. The order of releasing the buffers does not matter. We check for the byteBuf first just incase if we hit an exception while trying to acquire it, we know we don't need to check encodedByteBuf.

Comment (dhruve) [inline common/network-common/src/main/java/org/apache/spark/network/sasl/SparkSaslServer.java]: Had to fix a space, refactored the code. The linter however didn't complain.

**Comment by vanzin** [comment]
LGTM. Merging to master / 2.2 / 2.1.

**Comment by dhruve** [comment]
Thanks @vanzin

**Pull request closed without merging**